discovery/manager.go

```diff
@@ -153,41 +146,38 @@ func (m *Manager) cancelDiscoverers() {
 	for _, c := range m.discoverCancel {
 		c()
 	}
-	m.targets = make(map[poolKey][]*targetgroup.Group)
+	m.targets = make(map[poolKey]map[string]*targetgroup.Group)
 	m.discoverCancel = nil
 }
 
-func (m *Manager) addGroup(poolKey poolKey, tg []*targetgroup.Group) {
+func (m *Manager) updateGroup(poolKey poolKey, tgs []*targetgroup.Group) {
 	done := make(chan struct{})
 
 	m.actionCh <- func(ctx context.Context) {
-		if tg != nil {
-			m.targets[poolKey] = tg
+		if tgs != nil {
```

Comment:
Instead of checking `tgs` for nil here, you can do it at the very beginning of the method and return early (or even on the caller side).

Reply:
yeah we can check it here

```
		case tgs, ok := <-updates:
			// Handle the case that a target provider exits and closes the channel
			// before the context is done.
			if !ok || tgs ==nil {
				return
			}
```

but I am thinking if `updateGroup` is used by another caller which forgets to make the `nil` check?